web/pgadmin/utils/driver/psycopg3/server_manager.py

```
@@ -666,6 +671,26 @@ def create_connection_string(self, database, user, password=None):
+                    port=self.port,
+                    username=user,
+                    region=self.aws_region,
+                    profile=self.aws_profile if self.aws_profile else None,
+                    role_arn=self.aws_role_arn if self.aws_role_arn else None
+                )
+                # IAM authentication requires SSL
+                if not self.connection_params:
+                    self.connection_params = {}
+                if 'sslmode' not in self.connection_params:
+                    self.connection_params['sslmode'] = 'require'
+            except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)